src/prompt/primary.go

```diff
@@ -9,71 +9,71 @@ import (
 	"github.com/jandedobbeleer/oh-my-posh/src/terminal"
 )
 
-type ExtraPromptType int
-
-const (
-	Transient ExtraPromptType = iota
-	Valid
-	Error
-	Secondary
-	Debug
-)
-
 func (e *Engine) Primary() string {
-	if e.Config.ShellIntegration {
-		exitCode, _ := e.Env.StatusCodes()
-		e.write(terminal.CommandFinished(exitCode, e.Env.Flags().NoExitCode))
-		e.write(terminal.PromptStart())
-	}
-
-	// cache a pointer to the color cycle
-	cycle = &e.Config.Cycle
-	var cancelNewline, didRender bool
-
 	needsPrimaryRPrompt := e.needsPrimaryRPrompt()
 
-	for i, block := range e.Config.Blocks {
-		// do not print a leading newline when we're at the first row and the prompt is cleared
-		if i == 0 {
-			row, _ := e.Env.CursorPosition()
-			cancelNewline = e.Env.Flags().Cleared || e.Env.Flags().PromptCount == 1 || row == 1
+	switch e.Env.Shell() {
```

Comment:
no convinced that a switch is the best use-case for this one, why did you do in that direction?

Reply:
I'll reorganize the part of logic that adapts to PowerShell only.